src/context.cpp

```diff
@@ -382,42 +382,99 @@ namespace librealsense
     class dds_sensor_proxy : public software_sensor
     {
     public:
-        dds_sensor_proxy( std::string name, software_device* owner,
-                          std::shared_ptr< realdds::dds_device > const & dev) :
-            software_sensor(name, owner),
-            _dev(dev),
-            _name(name)
+        dds_sensor_proxy( std::string const & sensor_name,
+                          software_device * owner,
+                          std::shared_ptr< realdds::dds_device > const & dev )
+            : software_sensor( sensor_name, owner )
+            , _dev( dev )
+            , _name( sensor_name )
+        {
+        }
+
+        int add_dds_stream( std::shared_ptr< realdds::dds_stream > const & stream )
```

Comment:
Why `int` and not `size_t`? No need to cast

Reply:
It's the stream index, which is usually an `int` in libRS and the profile. We'll need to cast at some point. Might as well be here.

Reply:
You cast the sensor index at `to_rs2_video_stream`, I think this cast should be don't there too

Reply:
And this is not exactly how USB device indexes work, still need to work it out. Here you can have depth or color with index other than 0, and infrared with index 3 or 4. OK for this pull request, but needs more thinking